libycp/src/YCPMap.cc

```diff
@@ -178,7 +178,7 @@ YCPMapRep::compare(const YCPMap& m) const
 	{
 	    // equal length ==> pairwise comparison
 	    for( YCPMap::const_iterator pos_this = begin(), pos_m = m->begin();
-		 pos_this != end(), pos_m != m->end();
+		 pos_this != end();
```

Comment:
We have already checked that the containers are of equal size, so only one needs to be checked.